### DISCUSSION TITLE
Fleet window performance issues

### DISCUSSION THREAD

**Discussion by katepolak** [voxeltycoon/issues#619] [UI&UX]
**Describe the bug**
The fleet window has performance issues if a large number of vehicles is owned by the player. Attached is save file with over 240 vehicles that drops FPS from 60 to roughly 15 on my machine, with similar numbers reported by Xeteth#4013 on Discord who initially reported this.
In addition to the FPS drops the game hangs for roughly a second or two upon opening the window, with a smaller hang when closing it, but that one is just couple tenths of a second at most.

**Steps to reproduce**
Open the fleet window on  save with large number of vehicles.

**Build version**
0.85.0.24 STEAM

**Platform**
Various platforms

**Reproducibility**
Every time

**Logs/saves**

A save with over 240 vehicles that allows easy reproducibility, but probably isn't stricly necessary.
[p3sgzxu22t5.zip](https://github.com/voxeltycoon/bug-tracker/files/6328970/p3sgzxu22t5.zip)

**Comment by katepolak** [comment]
This has since been confirmed by another player in the Discord server, which rules out the already highly unlikely option of it being save specific.

**Comment by ovidiua2003** [comment]
I confirm it's happening to me as well. I noticed a small frame drop somewhere after 200+ vehicles. My main savem with 226 trucks and 12 trains,  and 60+ routes in total has a 26 FPS on a i7, with 2080RTX super (maxed out). The frame rate improves if i hide vehicle labels, but not by much, to 30.

Also when opening the  Fleet window, the FPS drops to 10 - 12 FPS.

But, I do love the game. It's a perfect modern take on TTD. Thank you for that.

**Comment by tbjoern** [comment]
Same for me. On my save I have ~200 Trucks, no Trains, and ~40 Routes. The game runs at 20-30 FPS (i5 4570, GTX 970). Opening the fleet window takes ~5 seconds and while the window is open the FPS drop significantly to 15 and less.

The first frame drops occur with ~50 trucks, before that I get stable 60 FPS (with and without fleet window).
